scripts/windows/restart-helpers.ps1

```diff
@@ -0,0 +1,64 @@
+# //////////////////////////////////////////////////////////////////////
+# restart-helpers.ps1
+
+function Test-Key([string] $path, [string] $key)
+{
+    return ((Test-Path $path) -and ((Get-Key $path $key) -ne $null))
+}
+
+function Remove-Key([string] $path, [string] $key)
+{
+    Remove-ItemProperty -path $path -name $key
+}
+
+function Set-Key([string] $path, [string] $key, [string] $value)
+{
+    Set-ItemProperty -path $path -name $key -value $value
+}
+
+function Get-Key([string] $path, [string] $key)
+{
+    return (Get-ItemProperty $path).$key
+}
+
+
+$global:restartKey = "Restart-And-Resume"
+function Clear-Any-Restart([string] $key=$global:restartKey)
+{
+    if (Test-Key $global:RegRunKey $key) {
+        Remove-Key $global:RegRunKey $key
+    }
+}
+
+$global:powershell = (Join-Path $env:windir "system32\WindowsPowerShell\v1.0\powershell.exe")
+function Restart-And-Resume([string] $script, [string] $step, [string] $action)
+{
+    Restart-And-Run $global:restartKey "$global:powershell $script -Step $step -Action $action"
+}
+
+function Prompt-Restart([string] $action)
+{
+    $answer = Read-Host -prompt "A restart is needed to continue $action. You can do this later if you prefer. Restart now? ([n]|y)"
```

Comment:
Yup, nicer. Next step is to put up a real live messagebox :)

Reply:
But that _does_ give you a real live messagebox! (If you run this in Powershell ISE rather than `powershell.exe`, that is) 😉 
![image](https://user-images.githubusercontent.com/3977982/116594747-69b77d80-a8d7-11eb-8222-bfcecf69b4bd.png)